projects/plugins/protect/src/js/components/admin-page/index.jsx

```diff
@@ -1,62 +1,35 @@
 import { AdminPage as JetpackAdminPage, Container } from '@automattic/jetpack-components';
-import { useProductCheckoutWorkflow } from '@automattic/jetpack-connection';
-import apiFetch from '@wordpress/api-fetch';
-import { useDispatch, useSelect } from '@wordpress/data';
+import { useConnection } from '@automattic/jetpack-connection';
 import { __ } from '@wordpress/i18n';
-import { addQueryArgs, getQueryArg } from '@wordpress/url';
-import React, { useEffect } from 'react';
-import { JETPACK_SCAN_SLUG } from '../../constants';
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import useNotices from '../../hooks/use-notices';
 import useWafData from '../../hooks/use-waf-data';
-import { STORE_ID } from '../../state/store';
-import InterstitialPage from '../interstitial-page';
 import Logo from '../logo';
 import Notice from '../notice';
 import Tabs, { Tab } from '../tabs';
 import styles from './styles.module.scss';
-import useRegistrationWatcher from './use-registration-watcher';
 
 const AdminPage = ( { children } ) => {
-	useRegistrationWatcher();
-
+	const { notice } = useNotices();
+	const { isRegistered } = useConnection();
 	const { isSeen: wafSeen } = useWafData();
-	const notice = useSelect( select => select( STORE_ID ).getNotice() );
-	const { refreshPlan, startScanOptimistically, refreshStatus, refreshScanHistory } =
-		useDispatch( STORE_ID );
-	const { adminUrl } = window.jetpackProtectInitialState || {};
-	const { run, isRegistered, hasCheckoutStarted } = useProductCheckoutWorkflow( {
-		productSlug: JETPACK_SCAN_SLUG,
-		redirectUrl: addQueryArgs( adminUrl, { checkPlan: true } ),
-		siteProductAvailabilityHandler: async () =>
-			apiFetch( {
-				path: 'jetpack-protect/v1/check-plan',
-				method: 'GET',
-			} ).then( hasRequiredPlan => hasRequiredPlan ),
-		useBlogIdSuffix: true,
-	} );
+	const navigate = useNavigate();
 
+	// Redirect to the setup page if the site is not registered.
 	useEffect( () => {
-		if ( getQueryArg( window.location.search, 'checkPlan' ) ) {
-			startScanOptimistically();
-			setTimeout( () => {
-				refreshPlan();
-				refreshStatus( true );
-				refreshScanHistory();
-			}, 5000 );
+		if ( ! isRegistered ) {
+			navigate( '/setup' );
 		}
-	}, [ refreshPlan, refreshStatus, refreshScanHistory, startScanOptimistically ] );
+	}, [ isRegistered, navigate ] );
```

Comment:
Added a unique `/setup` route for the interstitial page, and our query hooks remove the need for the `?checkPlan` rigamarole.